app/models/trip.rb

```diff
@@ -77,4 +86,23 @@ def self.total_by_year(years)
   def self.average_ride_duration
     average(:duration).round(2)
   end
+
+  def self.longest_ride
+    Trip.maximum(:duration)
+  end
+
+  def self.shortest_ride
+    Trip.minimum(:duration)
+  end
+
+  def self.most_frequent_zipcode
+    zipcode_id = Trip.group(:zipcode_id).count.max_by{|k,v| v}.first
```

Comment:
My only concern here is that just using k,v for key and value isn't quite descriptive enough. Might get dinged for one letter variables.